data/transactions/logic/assembler_test.go

```diff
@@ -3267,6 +3267,34 @@ add:
 	require.EqualError(t, err, "0: invalid syntax: not#pragma")
 }
 
+func TestAssembleImmediateRanges(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	/* Perhaps all of these "unable to parse" errors could be improved to
+	   discuss limits rather than bailout when the immediate is, in fact, an
+	   integer. */
+
+	testProg(t, "int 1; store 0;", AssemblerMaxVersion)
+	testProg(t, "load 255;", AssemblerMaxVersion)
+
+	testProg(t, "int 1; store -1000;", AssemblerMaxVersion,
+		Expect{1, "store unable to parse..."})
+	testProg(t, "load -100;", AssemblerMaxVersion,
+		Expect{1, "load unable to parse..."})
+	testProg(t, "int 1; store 256;", AssemblerMaxVersion,
+		Expect{1, "store i beyond 255: 256"})
```

Comment:
I had a gap in thinking through the error message: so failing to get type info w.r.t stack won't stop you from assembling, but what really stops you is in `asmDefault` where you call `byteImm` or `int8Imm`.

Does that sounds right about code path?

Reply:
Right, because the `type*` functions bailout and usually return `nil, nil, nil` when they are confused by the immediates or the stack size. They know that the real assembly routine will be angry, so they don't need to also complain.